node/src/components/rpc_server/rpcs/chain.rs

```diff
@@ -244,9 +253,18 @@ impl RpcWithOptionalParamsExt for GetBlockTransfers {
         api_version: ProtocolVersion,
     ) -> BoxFuture<'static, Result<Response<Body>, Error>> {
         async move {
+            // This RPC request is restricted by the block availability index.
+            let only_from_highest_contiguous_range = true;
+
             // Get the block.
             let maybe_block_id = maybe_params.map(|params| params.block_identifier);
-            let block_hash = match common::get_block(maybe_block_id, effect_builder).await {
+            let block_hash = match common::get_block(
+                maybe_block_id,
+                only_from_highest_contiguous_range,
+                effect_builder,
+            )
+            .await
+            {
```

Comment:
It should be fairly trivial to collapse these two requests into one.

Reply:
Yes, true. We might change the `GetBlockTransfers` to accept `BlockIdentifier` instead of `BlockHash`, but I'll prefer to do it in a separate PR, if really needed.

Or maybe you had some other idea in mind?

Reply:
I'd leave for 2.0 if we do want to collapse these requests.